src/Orckestra.Composer.Website/App_Code/OrdersHelper.cshtml

```diff
@@ -52,20 +59,40 @@
                         </p>
                     }
                     <br />
-                    <span class="compact-order-status"
-                          v-bind:class="{'text-warning' : order.OrderInfos.OrderStatusRaw == 'InProgress',
-                          'text-danger': order.OrderInfos.OrderStatusRaw == 'Canceled',
-                          'text-success': order.OrderInfos.OrderStatusRaw == 'Completed'}">
+                    <span v-if="order.OrderInfos.IsOrderPendingCancel" class="compact-order-status" v-bind:class="order.OrderInfos.OrderStatusTextCssClass">
+                        @Html.Localize("MyAccount", "T_PendingCancel") ( <a href="#" v-on:click.prevent="reload">@Html.Localize("MyAccount", "T_ReloadOrderStatus") </a>)
+                    </span>
+                    <span class="compact-order-status" v-bind:class="order.OrderInfos.OrderStatusTextCssClass" v-else="v-else">
                         {{ order.OrderInfos.OrderStatus }}
                     </span>
 
+
                 </div>
                 <div class="col-sm-4 text-sm-right">Total: <h5>{{order.OrderInfos.PricePaid}}</h5></div>
             </div>
         </div>
     </div>
 }
 
+@helper cancelOrderConfirmationModal()
+{
+    <div class="modal" id="cancelOrderModal" tabindex="-1" role="dialog">
+        <div class="modal-dialog  modal-sm" role="document">
+            <div class="modal-content">
+                <div class="modal-body">
+                    <button type="button" class="close" data-dismiss="modal" title="@Html.Localize("General", "B_Close")"><span class="fa fa-times-circle"></span></button>
+                    @Html.Localized("MyAccount", "L_ConfirmCancelOrder", "{{OrderNumber}}")
+                </div>
+                <div class="modal-footer">
+                    <button class="btn   btn-outline-secondary  modal--cancel" type="button">@Html.Localize("General", "B_Cancel")</button>
```

Comment:
Use here **B_No** as **B_Cancel** can confuse here
also add modal header and move close button inside, something like this:
  <div class="modal-header">
                    <h4 class="modal-title">Confirmation</h4>
                    <button type="button" class="close" data-dismiss="modal" title="@Html.Localize("General", "B_Close")"><span class="fa fa-times-circle"></span></button>
                </div>

just move text to localizations